runtime/lua/vim/uri.lua

```diff
@@ -52,18 +54,12 @@ do
   end
 end
 
----@private
-local function is_windows_file_uri(uri)
-  return uri:match('^file:/+[a-zA-Z]:') ~= nil
-end
-
 --- Get a URI from a file path.
 ---@param path string Path to file
 ---@return string URI
 local function uri_from_fname(path)
-  local volume_path, fname = path:match('^([a-zA-Z]:)(.*)')
-  local is_windows = volume_path ~= nil
   if is_windows then
+    local volume_path, fname = path:match('^([a-zA-Z]:)(.*)')
```

Comment:
Why move this? The PR description should say what problem is being solved. 

The old code handled `C:/` input the same on all systems. Unless that causes a problem, it's better than having platform-specific behavior.

Reply:
no problem to be solved . just thinking use an api to instead regex for improve .

Reply:
actually this not make too much sense.  close ?   `luv` use some low level to check system. maybe it not fast than regex ..